### DISCUSSION TITLE
Does a year of rehabilitation in QGEP/VSA model exists ?

### DISCUSSION THREAD

**Discussion by kandre** [QGEP/QGEP#745] [Q&A] [answered]
I would like to fill in the year of relining for some relined reaches/channels in our QGEP database. 

The existing attributes for reaches that accept a year are :
- wastewater_structure.year_of_construction : yaer when put into use
- wastewater_structure.year_of_replacement : year for which the lifetime of the structure is expected to be completed
- wastewater_structure.rv_base_year : reference year for computing the replacement value

To me those attributes cannot fit that need.

Does the VSA model and QGEP provides a way to store this information, like a year of rehabilitation ?
Does anyone knows where to store that information ?

**Comment by sjib** [comment, marked as answer]
There are four attributes for reliners:
reliner_material, reliner_nominal_size, relining_construction, relining_kind


and more generic the maintenance event that you can use for your case.

**Comment by kandre** [comment]
Yes, that's correct, and we already use those attributes for reliners. We could use the date of the maintenance event, however as far as I know, the model does not link relining with a specific maintenance event right ?

The question behind all that is that we also would like to compute something like the age of our network, and a year of rehabilitation should be used as an input.

**Comment by urskaufmann** [comment]
You can also use maintenance_events with status planned to define a date of next restoration.

And there should be a function that selects the last maintenance-event with a special kind, so it should be easy to get the last cleaning-, examination- or restoration-date.
